cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerClusterHealthService.java

```diff
@@ -165,6 +201,25 @@ private static Optional<DetailedServicesHealthCheck> getDetailedServicesHealthCh
         return Optional.empty();
     }
 
+    private Set<String> collectDisconnectedNodeManagers(List<ApiRole> allNMRoles) {
+        return allNMRoles.stream()
+                .filter(this::isNodeManagerDisconnected)
+                .filter(nm -> nonNull(nm.getHostRef()))
+                .map(nm -> nm.getHostRef().getHostname())
+                .collect(Collectors.toUnmodifiableSet());
+    }
+
+    private boolean isNodeManagerDisconnected(ApiRole role) {
```

Comment:
How does this API behave for nodes which are in STOPPED state? Or have just been started - but the NodeManager is not yet in RUNNING state ?

Reply:
As discussed, the test will be DISABLED once the roles are stopped. [Jira comment with additional details](https://jira.cloudera.com/browse/CB-21892?page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel&focusedCommentId=5994369#comment-5994369)